nixos/modules/services/networking/netbird.nix

```diff
@@ -1,171 +1,418 @@
     (mkIf cfg.enable {
-      # For backwards compatibility
-      services.netbird.tunnels.wt0.stateDir = "netbird";
+      services.netbird.clients.wt0 = {
+        port = mkDefault 51820;
+        suffix = mkDefault "";
+        interface = mkDefault "wt0";
+      };
     })
+    {
+      boot.extraModulePackages = optional
+        (cfg.clients != { } && (versionOlder kernel.version "5.6"))
+        kernelPackages.wireguard;
 
-    (mkIf (cfg.tunnels != { }) {
-      boot.extraModulePackages = optional (versionOlder kernel.kernel.version "5.6") kernel.wireguard;
+      environment.systemPackages = toClientList (client: client.wrapper);
 
-      environment.systemPackages = [ cfg.package ];
+      networking.dhcpcd.denyInterfaces = toClientList (client: client.interface);
+      networking.networkmanager.unmanaged = toClientList (client: "interface-name:${client.interface}");
 
-      networking.dhcpcd.denyInterfaces = attrNames cfg.tunnels;
+      systemd.network.networks = mkIf config.networking.useNetworkd (toClientAttrs (client:
+        nameValuePair "50-netbird-${client.interface}" {
+          matchConfig = {
+            Name = client.interface;
+          };
+          linkConfig = {
+            Unmanaged = true;
+            ActivationPolicy = "manual";
+          };
+        }
+      ));
 
-      systemd.network.networks = mkIf config.networking.useNetworkd (
-        mapAttrs'
-          (
-            name: _:
-            nameValuePair "50-netbird-${name}" {
-              matchConfig = {
-                Name = name;
-              };
-              linkConfig = {
-                Unmanaged = true;
-                ActivationPolicy = "manual";
-              };
-            }
-          )
-          cfg.tunnels
-      );
+      systemd.services = toClientAttrs (client:
+        nameValuePair client.name {
+          description = "A WireGuard-based mesh network that connects your devices into a single private network";
 
-      systemd.services =
-        mapAttrs'
-          (
-            name:
-            { environment, stateDir, ... }:
-            nameValuePair "netbird-${name}" {
-              description = "A WireGuard-based mesh network that connects your devices into a single private network";
+          documentation = [ "https://netbird.io/docs/" ];
 
-              documentation = [ "https://netbird.io/docs/" ];
+          after = [ "network.target" ];
+          wantedBy = optional client.autoStart "multi-user.target";
 
-              after = [ "network.target" ];
-              wantedBy = [ "multi-user.target" ];
+          path = optional (!config.services.resolved.enable) pkgs.openresolv;
 
-              path = with pkgs; [ openresolv ];
+          serviceConfig = {
+            ExecStart = "${getExe client.wrapper} service run";
+            Restart = "always";
 
-              inherit environment;
+            # see https://github.com/netbirdio/netbird/issues/1079#issuecomment-1823180247
```

Comment:
I fail to see how this issue is relevant to our case, you don't set `LogDirectory` as it is unused when using console and thus will not happen here